controller/telemetry/server.go

```diff
@@ -249,6 +214,27 @@ func (s *server) ListPods(ctx context.Context, req *read.ListPodsRequest) (*publ
 		return nil, err
 	}
 
+	// Reports is map from instance name to the absolute time of the most recent
+	// report from that instance.
+	reports := make(map[string]time.Time)
+	// Query Prometheus for reports in the last 30 seconds.
+	res, err := s.prometheusAPI.Query(ctx, "reports_total[30s]", time.Time{})
+	if err != nil {
+		return nil, err
+	}
+	if res.Type() != model.ValMatrix {
+		err = fmt.Errorf("Unexpected query result type (expected Matrix): %s", res.Type())
+		log.Errorf("%s", err)
+		return nil, err
+	}
+	for _, s := range res.(model.Matrix) {
+		labels := metricToMap(s.Metric)
+		timestamp := s.Values[len(s.Values)-1].Timestamp
+		secs := timestamp / 1000
+		nanos := (timestamp % 1000) * 1000000
+		reports[labels["pod"]] = time.Unix(int64(secs), int64(nanos))
```

Comment:
i think `s.Values[len(s.Values)-1].Timestamp` is already a `time.Time`?

Reply:
it's a `model.Time` from Prometheus, which is a type alias for `int64`.